src/runtime/task/task_engine.cpp

```diff
@@ -288,4 +291,49 @@ void task_engine::get_queue_info(/*out*/ std::stringstream &ss)
         }
     }
 }
-} // end namespace
+
+void task_engine::register_cli_commands()
+{
+    static std::once_flag flag;
+    std::call_once(flag, [&]() {
+        _task_queue_max_length_cmd = dsn::command_manager::instance().register_command(
+            {"task.queue_max_length"},
+            "task.queue_max_length <pool_code> [queue_max_length]",
+            "get/set the max task queue length of specific thread_pool, -1 means no limit",
+            [this](const std::vector<std::string> &args) {
+                if (args.size() < 1) {
+                    return std::string("ERR: invalid arguments, task.queue_max_length <pool_code> "
+                                       "[queue_max_length]");
+                }
+
+                for (auto &it : _pools)
+                    if (it) {
+                        if (it->_spec.pool_code.to_string() == args[0]) {
+                            if (args.size() == 1) {
+                                return fmt::format("task queue {}, length {}",
+                                                   args[0],
+                                                   it->_spec.queue_length_throttling_threshold);
+                            }
+                            if (args.size() == 2) {
+                                int queue_length = INT_MAX;
+                                if ((args[1] != "INT_MAX") &&
+                                    (!dsn::buf2int32(args[1], queue_length))) {
+                                    return fmt::format(
+                                        "queue_max_length must >= 0, or set `INT_MAX`");
```

Comment:
This comment makes me confused, it seems that the `queue_length` can not be parsed as int32, and current implementation allow this value to be set as -1. 
Besides, I have a question. Is the length as no_limit equal to TM_NONE?

Reply:
Change the helper